### ISSUE TITLE
The average color seems to have broken with the latest update.

### ISSUE THREAD

**Issue by tchesket** [samclane/LIFX-Control-Panel#34]
When testing about the latest beta (1.6.0 beta 5), the average color feature seems to be active very strange. After I set up the rectangle for each light, and turn each bulbs average color feature on, it seems like it works for a second, and then it starts doing weird things which I'm not exactly sure how to describe. I am using microsoft paint to set each half of my screen to a certain color, light red on the left and green on the right. I will turn on the avg color(s), then open paint and f11 to full screen. It seems like it gets the colors right at first (although the accuracy seems to have been affected some). But then when I move the paint window around, or change the colors in the paint program, the lights stop matching up with what is on the screen. Like, after moving the paint window, the left light went from red (correct) to a bluish color, and the right light went from the green (correct) to I think the same weird color as the other bulb. It then acts strange until I stop the avg color and restart it, which causes the issue to repeat in exactly the same way. Here is my log file, I hope it helps.

[lifx-control-panel.log](https://github.com/samclane/LIFX-Control-Panel/files/2819788/lifx-control-panel.log)

**Comment by tchesket** [comment]
The problem is also reproducible with 1.6 beta 4. Also, 1 thing I forgot, the lights now seem to change color slightly sometimes even when there is nothing changing on the screen.

**Comment by tchesket** [comment]
It also seems to be that when I make the paint window 1 solid color, it more or less correctly grabs the color. But it is applying the color to both lights, regardless of whether or not the avg color feature is running on just 1 or on both lights. And the responsiveness in changing colors is noticeably worse in general. The problem resolves itself if I run one of the older releases, but keeps appearing again with the new ones. Let me know if I can help out anymore. Sorry I haven't been on much, my free public wifi that I mentioned has not been working for several days.

**Comment by tchesket** [comment]
Did you perhaps change the way in which the rectangles are drawn/created in the programming? I'm noticing something else now which is kind of strange, but I was able to get some of the other programs we have talked about working and tested them out; it seems that my computer is doing something strange when programs try to get my screen size/resolution. For example, one of the other programs has a tool that lets you select your screen area ("rectangles") by opening a "snapshot" window, and then you are supposed to draw a bow around the part of your screen that you want to be covered (a process which I find quite annoying, I much prefer your method of manual entry). But when that window opens, I assume it is supposed to give you a window showing your whole screen, but it crops or somehow distorts my screen and only shows me a portion (perhaps the left half? or even less? I'm not sure) of my whole screen area. And it is zoomed in or otherwise stretched in weird ways. It's like my computer is reporting that my resolution is something different (lower, with a different aspect ratio) than it actually is. This might just be a problem specific to my machine, but I think this might have something to do with the strange new behavior I'm getting. Let me know if that didn't make sense and I will clarify.

**Comment by tchesket** [comment]
Well I just have the one monitor. A tv really, going through hdmi to my geforce graphics card (drivers are always kept up-to-date). And its not just that the colors are "less accurate" with the newest 2 betas, but they are totally wrong, and they act funny. Like I will have red and green on the screen and the lights will both randomly fade to slightly different shades of blue or something, and then get stuck there and will stop responding to any screen changes. Stuff like that

**Comment by samclane** [comment]
Ok, I tried with the latest release (Beta 6?) and with the latest stable release (1.5.4). I made a subregion (200, 200, 400, 400), opened paint, made the background orange, and started dragging a blue square in-and-out of the field. I noticed on the new release that I'd occasionally get big flickers of white when first moving the square _into_ the zone. The older release seemed to be more stable and consistent with its color changes, even if they're not as smooth. I was also getting some weird colors too. 

This may be due to the color changes being non-contiguous in HSBK color-space. Or maybe I should be averaging across the `Hue` parameter, just the saturation and brightness. That would reduce flicker while keeping the correct chromatic feature. 

I'll screw around with it some more.

**Comment by tchesket** [comment]
Okay I will test it out. Thanks. With my setup it was extremely obvious, I have a 4way light fixture/ceiling fan, and I have 1 bulbs in the left and 1 in the right, and I am using a 4k tv as a monitor, setting the left light to (0, 1920, 0, 2160) and the right bulb to (1920, 3840, 0, 2160), as to capture each half of the screen. This seems like the most optimal way to set it up to me. I would like to get a 3rd bulb and place it in between the two facing the tv/wall and probably just do something like (1280, 2560, 0, 2160).

The program doesn't have support for the lifx light strip/24 zones yet does it? I'm thinking about buying one with my tax return and putting it behind my TV, but I'm not sure what kind of results I'd be able to get with it. From what I have seen, the Philips lights seem more suited to that kind of thing but their lights suck in terms of brightness.

**Comment by samclane** [comment]
[Try this link to Beta 6](https://github.com/samclane/LIFX-Control-Panel/releases/tag/1.6.0-beta6). I think the Hue change really fixed the problem. 

Also, no, the program doesn't support anything more complex than an RGBW bulb. Developing on hardware you don't personally have can be pretty time consuming. I only had 1 bulb for a bit, so I had to simulate the others in software, and I had to rely on another Reddit user to validate. When I did get my second bulb, I immediately found like 2/3 major issues that made the program almost unusable. I was able to fix it quickly with the actual hardware present. I know it sounds selfish, but I want to prioritize features that I can personally confirm work.
